src/Serializer.ts

```diff
@@ -6,7 +6,7 @@ export const encode = (value: any, indent?: number | undefined) => {
   return JSON.stringify(
     value,
     (key, value) => {
-      if (typeof value[BOXED_TYPE] === "string") {
+      if (typeof value?.[BOXED_TYPE] === "string") {
```

Comment:
let's maybe rather use the following to maximize browser compatibility

```
if (value == null) {
  return value
}
```

Reply:
It's been a while since I had to think about syntax support of older browsers! 😅 

I had a niggling that the Typescript compiler actually compiled out optional chaining - depending on the build target - and ran a test build. It did indeed turn the optional chaining into the following:
```ts
null==n?n:"string"==typeof n[o]
```

Then to _really_ satisfy my curiosity, I tried your suggestion and it actually compiled to the exact same output!